src/ftl/ast/parser.js

```diff
@@ -2,808 +2,104 @@
-
-    // We ignore the first white space of each line
-    if (this._source[this._index] === ' ') {
-      this._index++;
-    }
-
-    let eol = this._source.indexOf('\n', this._index);
-
-    let content = this._source.substring(this._index, eol);
-
-    while (eol !== -1 && this._source[eol + 1] === '#') {
-      this._index = eol + 2;
-
-      if (this._source[this._index] === ' ') {
-        this._index++;
-      }
-
-      eol = this._source.indexOf('\n', this._index);
-
-      if (eol === -1) {
-        break;
-      }
-
-      content += `\n${this._source.substring(this._index, eol)}`;
-    }
-
-    if (eol === -1) {
-      this._index = this._length;
-    } else {
-      this._index = eol + 1;
-    }
-
-    return new AST.Comment(content);
-  }
-
-  error(message, start = null) {
-    const pos = this._index;
-
-    if (start === null) {
-      start = pos;
-    }
-    start = this._findEntityStart(start);
-
-    const context = this._source.slice(start, pos + 10);
-
-    const msg =
-      `\n\n  ${message}\nat pos ${pos}:\n------\n…${context}\n------`;
-    const err = new L10nError(msg);
-
-    const row = this._source.slice(0, pos).split('\n').length;
-    const col = pos - this._source.lastIndexOf('\n', pos - 1);
-    err._pos = {start: pos, end: undefined, col: col, row: row};
-    err.offset = pos - start;
-    err.description = message;
-    err.context = context;
-    return err;
-  }
-
-  getJunkEntry() {
-    const pos = this._index;
-
-    let nextEntity = this._findNextEntryStart(pos);
-
-    if (nextEntity === -1) {
-      nextEntity = this._length;
-    }
-
-    this._index = nextEntity;
-
-    let entityStart = this._findEntityStart(pos);
-
-    if (entityStart < this._lastGoodEntryEnd) {
-      entityStart = this._lastGoodEntryEnd;
-    }
-
-    const junk = new AST.JunkEntry(
-      this._source.slice(entityStart, nextEntity));
-    return junk;
-  }
-
-  _findEntityStart(pos) {
-    let start = pos;
-
-    while (true) {
-      start = this._source.lastIndexOf('\n', start - 2);
-      if (start === -1 || start === 0) {
-        start = 0;
-        break;
-      }
-      const cc = this._source.charCodeAt(start + 1);
-
-      if (isIdentifierStart(cc)) {
-        start++;
-        break;
-      }
-    }
-
-    return start;
-  }
-
-  _findNextEntryStart(pos) {
-    let start = pos;
-
-    while (true) {
-      if (start === 0 ||
-          this._source[start - 1] === '\n') {
-        const cc = this._source.charCodeAt(start);
-
-        if (isIdentifierStart(cc) || cc === 35 || cc === 91) {
-          break;
-        }
-      }
-
-      start = this._source.indexOf('\n', start);
-
-      if (start === -1) {
-        break;
-      }
-      start++;
-    }
-
-    return start;
-  }
+  return new AST.Pattern(elements, false);
 }
 
 export default {
-  parseResource: function(string, { withSource = true } = {}) {
-    const parser = new Parser(withSource);
-    return parser.getResource(string);
-  },
-};
+  parse
```

Comment:
You can also `export function parse(source) {` at the top of the file and remove this.